Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -774,6 +777,26 @@ bool FullscreenManager::isSimpleFullscreenDocument() const
     return foundFullscreenFlag;
 }
 
+Document* FullscreenManager::deepestFullscreenDocument() const
+{
+    auto* fullscreenElement = this->fullscreenElement();
+    if (!fullscreenElement)
+        return nullptr;
+
+    if (!is<HTMLFrameOwnerElement>(fullscreenElement))
+        return m_document.ptr();
+
+    auto* contentDocument = downcast<HTMLFrameOwnerElement>(*fullscreenElement).contentDocument();
+    if (!contentDocument)
+        return m_document.ptr();
+
+    auto* fullscreenDocument = contentDocument->fullscreenManager().deepestFullscreenDocument();
```

Comment:
"deepest" seems like the wrong description. "deepest" implies "leaf-most" but I think this is "outer fullscreen document"?

Reply:
Also, since this crosses frame boundaries, and it seems like `FullscreenManager` is a per-Document thing, this function should live on Page, and may have to be different in future for site isolation.

Reply:
Can't just be `currentFullscreenDocument()` because multiple documents are in fullscreen simultaneously (specifically all the ancestor documents of the "outermost" one).

Can't just be `topmostFullscreenDocument()` because that's too close to `topDocument()` and implies a heirarchy that goes in a different direction.

So, I'd be happy to change this to `outermostFullscreenDocument()`. And I'll move it into Page.